rpcs3/util/vm_native.cpp

```diff
@@ -362,12 +362,21 @@ namespace utils
 
 	u8* shm::map_self(protection prot)
 	{
-		if (!m_ptr)
+		void* ptr = m_ptr;
+
+		while (!ptr)
 		{
-			m_ptr = this->map(nullptr, prot);
+			const auto mapped = this->map(nullptr, prot);
+
+			if (m_ptr.compare_exchange(ptr, mapped))
+			{
+				break;
+			}
+
+			this->unmap(mapped);
```

Comment:
It doesn't make sense to discard map on failure

Reply:
If there are more mappings left allocated the destructor may not work properly, also for consistency of address value when this function is called concurrently.

Reply:
Nvm, you misunderstood my comment on old version.